collectors/VCenterPropertiesCollector.py

```diff
@@ -0,0 +1,53 @@
+from BaseCollector import BaseCollector
+import os, time
+from prometheus_client.core import GaugeMetricFamily
+from tools.Resources import Resources
+from tools.YamlRead import YamlRead
+
+
+class VcenterPropertiesCollector(BaseCollector):
+    def __init__(self):
+        self.iteration = 0
+        while not self.iteration:
+            time.sleep(5)
+            self.get_iteration()
+            print("waiting for initial iteration")
+        print("done: initial query")
+        self.property_yaml = YamlRead('collectors/property.yaml').run()
+
+    def collect(self):
+        if os.environ['DEBUG'] >= '1':
+            print('vCenterPropertiesCollector starts with collecting the properties')
+
+        g = GaugeMetricFamily('vrops_vcenter_properties', 'testtest',
+                              labels=['datacenter', 'vcenter', 'propkey'])
```

Comment:
According to our latest findings, it is useful to insert a method that describes the collector. Please adopt this also in your collectors. To do this, you need to initialize the GaugeMetrik object in the init method. Make sure that you reference this object everywhere.

```
def __init__(self):
         self.wait_for_inventory_data()
         self.property_yaml = YamlRead('collectors/property.yaml').run()
         self.g = GaugeMetricFamily('vrops_vcenter_properties', 'testtest',
                                   labels=['datacenter', 'vcenter',  'propkey'])

    def describe(self):
        yield self.g
```